feat: Hide keyboard shortcuts on mobile screens

tofumatt opened the pull request:
Fix #9050.

## Description
Hide the shortcuts for mobile screens.

## How has this been tested?
Tested locally; works.

## Screenshots
![screen shot 2018-08-16 at 14 31 03](https://user-images.githubusercontent.com/90871/44211903-f76fd080-a161-11e8-8e36-498780731587.png)

## Types of changes
CSS change.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->

tofumatt committed:
feat: Hide keyboard shortcuts on mobile screens

Fix #9050.

tofumatt committed:
docs: Clarify reasoning for wanting more space

jasmussen commented:
I'm actually seeing some issues, though I'm not sure if that's a result of this PR or a regression from the one I just merged :(

This PR:

![issues](https://user-images.githubusercontent.com/1204802/44217215-3fe5b900-a177-11e8-9485-68ef1876b651.gif)

Master:

![master](https://user-images.githubusercontent.com/1204802/44217223-42e0a980-a177-11e8-9548-55798f443792.gif)

Upon further investigation, yeah definitely looks like a regression from my PR, sorry :( Pushing a fix in a second.

jasmussen committed:
Add min-width to menu.

This fixes a regression I introduced in https://github.com/WordPress/gutenberg/pull/8756/files#diff-dcb8eefee4aaf82c9ddb042494547527R158.

jasmussen commented:
Just pushed a fix to the issue:

![fixed](https://user-images.githubusercontent.com/1204802/44217522-f21d8080-a177-11e8-99d9-97fe44b51b3c.gif)

See also:

![screen shot 2018-08-16 at 17 13 15](https://user-images.githubusercontent.com/1204802/44217532-f6499e00-a177-11e8-9798-fcb7268a8e55.png)

![screen shot 2018-08-16 at 17 13 19](https://user-images.githubusercontent.com/1204802/44217536-f8136180-a177-11e8-816f-37566cf03359.png)

jasmussen reviewed (approved):
This improves on the menu situation on small viewports, and it also helps (hot)fix a regression I introduced. Sorry about that. Let's get this into 3.6.

youknowriad commented on packages/components/src/menu-item/style.scss:
```diff
@@ -51,9 +51,17 @@
+	// Hide the keyboard shortcuts on mobile, because they aren't super-useful
+	// for most mobile users and it frees up screen space for any item
+	// with a long description.
+	display: none;
```
Probably fine for now, but I wonder if this should be hidden entirely in the generic MenuItem component or something specific to the Global menu item. 

It can be used in the Block Settings Menu and elsewhere since it's a generic component.

tofumatt commented:
I think hidden globally makes sense. Sorry, that was deliberate, I should’ve mentioned. My patch is global, right? 😅

Most mobile viewports happen on devices with keyboards that don’t support shortcuts, so hiding that information entirely to reduce noise and also give more space to other info on a screen with reduced real estate seems a good rule globally to me.

- Matt (Sent from mobile)

> On 16 Aug 2018, at 16:20, Riad Benguella <notifications@github.com> wrote:
> 
> @youknowriad commented on this pull request.
> 
> In packages/components/src/menu-item/style.scss:
> 
> >  	padding-left: 8px;
> +
> +	// Hide the keyboard shortcuts on mobile, because they aren't super-useful
> +	// for most mobile users and it frees up screen space for any item
> +	// with a long description.
> +	display: none;
> Probably fine for now, but I wonder if this should be hidden entirely in the generic MenuItem component or something specific to the Global menu item.
> 
> It can be used in the Block Settings Menu and elsewhere since it's a generic component.
> 
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub, or mute the thread.
